sympy/assumptions/tests/test_query.py

```diff
@@ -2269,6 +2278,24 @@ def Symbol(expr, assumptions):
             if expr in conjuncts(assumptions):
                 return True
     register_handler('mersenne', MersenneHandler)
-
     n = Symbol('n', integer=True)
+    assert ask(Q.mersenne(7))
     assert ask(Q.mersenne(n), Q.mersenne(n))
+    del Q.mersenne
+
+
+    # New handler system
+    MersenneHandler2 = AskHandlerClass('MersenneHandler2')
+    register_handler('mersenne2', MersenneHandler2)
+    @Q.mersenne2.handler.register(Integer)
```

Comment:
Why do I need to create an instance of `AskHandlerClass` here? It doesn't seem necessary since I don't use it for anything.

Presumably this could just be `Q.add('mersenne2')`.

Reply:
You're right

Reply:
I introduced a simple method, `Q.add_predicate` which works same as your suggestion. Perhaps this can replace `register_handler`. How do you think?

Reply:
Adding a method somehow intervened into SAT system and failed the test. I'm trying with some other approach..

Reply:
I introduced `generate_predicate` function which is designed for new handler design. `register_handler` and `remove_handler` are no longer used, and will be deprecated in succeeding PRs.